Marlin/src/MarlinCore.cpp

```diff
@@ -1155,79 +1155,79 @@ void setup() {
     #define INIT_CUSTOM_USER_BUTTON_PINS(N) (PIN_EXISTS(USER_GCODE_PIN_##N) && ENABLED(USER_GCODE_PIN_INIT_##N) && defined(USER_GCODE_PIN_STATE_##N))
     #define INIT_CUSTOM_USER_BUTTON_PIN(N) do{ SET_INPUT(USER_GCODE_PIN_##N); WRITE(USER_GCODE_PIN_##N, !USER_GCODE_PIN_STATE_##N); }while(0)
 
-    #if INIT_CUSTOM_USER_BUTTON_PINS(1)
```

Comment:
The idea here was to allow the user to not init the PIN at boot time (USER_GCODE_PIN_INIT_##N, might be useful if the PIN is already part of another feature) and to ensure proper configuration settings (which are not yet checked in HAS_CUSTOM_USER_BUTTON macro.

Reply:
I removed INIT_CUSTOM_USER_BUTTON_PINS(N) macro and USER_GCODE_PIN_INIT_n. That way USER_GCODE_PIN_n will always be initiated to ensure proper state